src/TicketValidator.java

```diff
@@ -1,12 +1,11 @@
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.*;
 
 class TicketValidator {
     private int totalTickets;
     private int validTickets;
-    private Map<String, Integer> violationsCount;
+    private final Map<String, Integer> violationsCount;
```

Comment:
Why final?

Reply:
the violationsCount map is final to ensure the reference to the map never changes for this project, but i  understand that for this variable for future it's not good practice